File: medium/62-unique-paths/dp.go
// https://leetcode.com/problems/unique-paths/

package uniquePaths

func uniquePaths(m int, n int) int {
	dp := make([][]int, m)

	for i := 0; i < m; i++ {
		row := make([]int, n)
		dp[i] = row
		for j := 0; j < n; j++ {
			if i == 0 && j == 0 {
				dp[i][j] = 1
			}
			if j > 0 {
				dp[i][j] += dp[i][j-1]
			}
			if i > 0 {
				dp[i][j] += dp[i-1][j]
			}
		}
	}

	return dp[m-1][n-1]
}

File: medium/62-unique-paths/dp.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [[0 for _ in range(n)] for _ in range(m)]

        dp[0][0] = 1
        i = 0
        while i < m:
            j = 0
            while j < n:
                if i - 1 >= 0:
                    dp[i][j] += dp[i-1][j]
                if j - 1 >= 0:
                    dp[i][j] += dp[i][j-1]
                j += 1
            i += 1

        return dp[m-1][n-1]